### DISCUSSION TITLE
how to start?

### DISCUSSION THREAD

**Discussion by whyAllan** [Game-as-a-Service/Pandemic-Reign-of-Cthulhu#44] [Q&A] [answered]
Hi there I come cause a mesage that I saw in a discord group, I would like very much to help in this project my question is how to start? I have more knowledge in python and backend but I also have some experience with front-end including typescrypt but no experience working with others, any light would be great, thks

**Comment by metalalive** [comment, marked as answer]
Hi Trepalan, and everyone interested with the project development

You can check the open issues in [GitHub Issue](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues) , if you find anything that is pending or nobody works on it , you can discuss with me (for backend development) or @wingtkw (for frontend development) for the software design and implementation.

### Briefly describe current situation
#### A lot of work in frontend dev,
[Issue #39](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues/39)
  - we've done with [part of build script](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/blob/main/.github/workflows/frontend_ci.yml)  but have not added test cases yet.
  - Testing is required because we'd like to apply [TDD](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu#contribution)  to the app development
  - Here is previous discussion about [frontend CI flow](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/discussions/35) , Maybe you can discuss with @wingtkw  about the frontend test cases to add.

[issue #53](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues/17)
- how to store game status (retreived data from backend) at front-end app , the discussion is ongoing.

#### Backend dev
- [issue #12](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues/12), currently it is pending , we need a API endpoint to transition game state (from here you may need to read [the manual](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu#overview) to understand the game rule).
- [issue #37](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues/37) , I am currently working on the implementation of  pub/sub pattern, using `socket.io` and `python-socketio` library,
  - see [the recap of previous discussion](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/discussions/29).
  - also there is [TODO list](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/issues/37#issuecomment-1842635038) for the issue 12


### Before you start coding
Read the [contribution section](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu#contribution) first to understand the software-engineering techniques we (core maintainers and contributors) would like to apply to the game app.

Part of previous discussion was done in Chinese, if you are interested with the discussion but don't fully understand that , please let me know.

That's it, please leave your comment for any question, thanks

**Reply by metalalive** [reply]
Hi, everyone, if you think my answer of this discussion thread can be improved,
Feel free to leave message , appreciate any feedback, thanks

**Comment by metalalive** [comment]
@liamlin1102 feel free to start conversation in any issue above you'd like to work with,
this discussion thread might be good for starter
